filebeat/input/tcp/client.go

```diff
@@ -0,0 +1,116 @@
+package tcp
+
+import (
+	"bufio"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/filebeat/harvester"
+	"github.com/elastic/beats/filebeat/util"
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// Client is a remote client.
+type Client struct {
+	conn           net.Conn
+	forwarder      *harvester.Forwarder
```

Comment:
can we replace `*harvester.Forwarder` by an interface or function type? Some more isolation from harvester.Forwarder (I hope to remove Forwarder and Outlet in the future).

Reply:
Knowing that it make sense to change it.

Reply:
Similar to #6439  ?:)
I was planning to do it in second part but I can do it right now.

Reply:
let's keep the PR as small as possible and do the interface when we have a second implementation.